src/main.ts

```diff
@@ -7,15 +7,65 @@ async function run() {
   try {
     const token = core.getInput('repo-token', {required: true});
     const configPath = core.getInput('configuration-path', {required: true});
+    const notFoundLabel = core.getInput('not-found-label');
+    const operationsPerRun = parseInt(
+      core.getInput('operations-per-run', {required: true})
+    );
+    let operationsLeft = operationsPerRun;
+
+    const client = new github.GitHub(token);
 
     const prNumber = getPrNumber();
-    if (!prNumber) {
-      console.log('Could not get pull request number from context, exiting');
+    if (prNumber) {
+      await processPR(client, prNumber, configPath, notFoundLabel);
       return;
     }
 
-    const client = new github.GitHub(token);
+    const opts = await client.pulls.list.endpoint.merge({
+      owner: github.context.repo.owner,
+      repo: github.context.repo.repo,
+      state: 'open',
+      sort: 'updated'
+    });
+
+    let prs = await client.paginate(opts);
+    prs = prs.filter(pr => {
+      const hasLabels = pr.labels && pr.labels.length > 0;
+      if (hasLabels) {
+        core.debug(
+          `pr ${pr.number} already has ${pr.labels.length} labels`
+        );
+      }
+      return !hasLabels;
+    });
+
+    for (const pr of prs) {
+      core.debug(`performing labeler at pr ${pr.number}`);
+      if (operationsLeft <= 0) {
+        core.warning(
+          `performed ${operationsPerRun} operations, exiting to avoid rate limit`
+        );
+        return 0;
+      }
 
+      if (await processPR(client, pr.number, configPath, notFoundLabel)) {
+        operationsLeft -= 1;
+      }
+    };
+  } catch (error) {
+    core.error(error);
+    core.setFailed(error.message);
+  }
+  return;
```

Comment:
Isn't `async function run()` return a `Promise<void>` ? the 2 return statements change it.

Reply:
L39 is return for filter

Reply:
My bad for the filter :)

Reply:
Done.

Removed catch from processPR too, otherwise a fail to label will be only a warning.